### ISSUE TITLE
refreshToken is missing sometimes.

### ISSUE THREAD

**Issue by gazialankus** [dart-lang/googleapis_auth#23]
Why do we have to have the refresh token here? 
https://github.com/dart-lang/googleapis_auth/blob/master/lib/src/auth_http_utils.dart#L81

```
  AutoRefreshingClient(Client client, this.clientId, this.credentials,
      {bool closeUnderlyingClient: false})
      : super(client, closeUnderlyingClient: closeUnderlyingClient) {
    assert(credentials.refreshToken != null); //<--this
    authClient = authenticatedClient(baseClient, credentials);
  }
```

Here's what I'm doing: 

1. I'm getting an authenticated client with `clientViaUserConsentManual()` 
2. I'm doing this call `await new GamesManagementApi(client).achievements.resetAllForAllPlayers();` and getting this in an exception: `error=insufficient_scope, scope="https://www.googleapis.com/auth/games`, which crashes my app. 
3. From then on, `clientViaUserConsentManual()` stops working in subsequent executions. I'm hitting on the assert that I'm questioning above, as refresh_token is not there in the response. Only by removing that assert I am able to test one more time. 

So, why do we have that assert and do I have any other option but to remove it?

**Comment by mkustermann** [comment]
The `clientViaUserConsentManual()` creates always an `AutoRefreshingClient` which requires the refresh token (hence the assert) because it will refresh the access token (using the refresh token) once it expired.

If you know the token doesn't expire anytime soon, you can also obtain the credentials separately, something like this:
```
  http.Client baseClient = new http.Client();
  AccessCredentials credentials = await obtainAccessCredentialsViaUserConsentManual(..., baseClient, ...);

  // Instead of this
  AuthClient authClient = autoRefreshingClient(clientId, baseClient, credentials);  

  // Use this
  AuthClient authClient = authenticatedClient(baseClient, credentials);
```

Though I'm not sure why you don't get a refresh token in the first place.

**Comment by gazialankus** [comment]
I figured out my mistake that was causing the `insufficient_scope` exception I talked about above. I was creating the auth code before the call to `clientViaUserConsentManual()`. Probably the crypto includes some kind of timestamp check. In this specific situation I was not getting a refresh token in subsequent calls if I used the same user. 

Anyway, correct usage probably would not have such a problem. I'll just close this for now. Thank you for the explanation about the AutoRefreshingClient and the sample code, much appreciated!
